packages/block-editor/src/components/autocomplete/index.js

```diff
@@ -27,41 +26,45 @@ import blockAutocompleter from '../../autocompleters/block';
 const EMPTY_ARRAY = [];
 
 /**
- * Wrap the default Autocomplete component with one that
- * supports a filter hook for customizing its list of autocompleters.
+ * Wrap the default Autocomplete component with one that supports a filter hook
+ * for customizing its list of autocompleters.
  *
- * This function is exported for unit test.
- *
- * @param  {Function} Autocomplete Original component.
- * @return {Function}              Wrapped component
+ * @type {import('react').FC}
  */
-export function withFilteredAutocompleters( Autocomplete ) {
-	return ( props ) => {
-		let { completers = EMPTY_ARRAY, blockName } = props;
-		completers = useMemo( () => {
-			if ( blockName === getDefaultBlockName() ) {
-				return completers.concat( [ blockAutocompleter ] );
-			}
-			return completers;
-		}, [ completers, blockName ] );
+function BlockEditorAutocomplete( props ) {
+	const { name } = useBlockEditContext();
+
+	let { completers = EMPTY_ARRAY } = props;
+
+	completers = useMemo( () => {
```

Comment:
Note: There's a bit more intensive refactoring here than just a direct port. I observed the prior logic was fairly unoptimized if `editor.Autocomplete.completers`, since it would create a new copy of the array on every render. It's also not necessary to consider the filtering on every render, if we're already managing this memoization.